lib/ssl/src/ssl_handshake.erl

```diff
@@ -2209,24 +2203,24 @@ bad_key(#{algorithm := ecdsa}) ->
     unacceptable_ecdsa_key.
 
 cert_status_check(_,
-                  #{ocsp_state := #{ocsp_stapling := true,
-                                    ocsp_expect := stapled}},
+                  #{stapling_state := #{configured := true,
+                                        status := stapled}},
                   _VerifyResult, _, _) ->
     %% OCSP staple will now be checked by
     %% ssl_certificate:verify_cert_extensions/2 in ssl_certificate:validate
     valid;
 cert_status_check(OtpCert,
-                  #{ocsp_state := #{ocsp_stapling := false}} = SslState,
+                  #{stapling_state := #{configured := false}} = SslState,
                   VerifyResult, CertPath, LogLevel) ->
     maybe_check_crl(OtpCert, SslState, VerifyResult, CertPath, LogLevel);
 cert_status_check(_OtpCert,
-                  #{ocsp_state := #{ocsp_stapling := true,
-                                    ocsp_expect := undetermined}},
+                  #{stapling_state := #{configured := true,
+                                        status := undetermined}},
                   _VerifyResult, _CertPath, _LogLevel) ->
     {bad_cert, {revocation_status_undetermined, not_stapled}};
 cert_status_check(_OtpCert,
-                  #{ocsp_state := #{ocsp_stapling := true,
-                                    ocsp_expect := no_staple}},
+                  #{stapling_state := #{configured := true,
+                                        status := no_staple}},
                   _VerifyResult, _CertPath, _LogLevel) ->
     {bad_cert, {revocation_status_undetermined, not_stapled}}.
```

Comment:
fallback to CRL check if OSCP staple(s) was not received?

Reply:
To be able to fallback to CRL check, CRL options must have been specified so it can not be done unconditionally.

Reply:
I'm not sure how this CRL fallback should look like. Maybe we can take that discussion after merging this PR ...

Reply:
I think it will look like when OCSP check finds itself in result "undetermined" it should call mayb_check_crl if crl_check is set to relevant value  (not false, and not peer if CA cert in TLS-1.3, only peer relevant in TLS-1.2) to see if it can get a valid result.  It would have to be handled in different places for TLS-1.3 (certificate callback) and it can be handled in cert_status_check for TLS-1.2.  We can fix it in follow up PR for master, as we really like to merge this soon to be able to also continue work with other TLS refactor jobs.

Reply:
I created backlog item for finalizing this thread.
Resolving it here.